### PR TITLE
[#49] Set up Firebase Distribution for Android

### PR DESCRIPTION
**Pull request by chornerman**
#49

## What happened 👀

Set up Firebase Distribution for Android

## Insight 📝

We will deploy staging debug when pushing to `develop` or `release/**` (merge feature branches) and deploy production debug when pushing to `main` (merge release branches)
- [x] Create and set up projects on the Firebase web client for both staging and production
- [x] Add Firebase dependencies and increase Android's minimum SDK (required for Firebase to work)
- [x] Encrypt `.env` files using GPG
- [x] Encrypt `google-services.json` using GPG
- [x] Add script to decrypt the encrypted files (will be used in the deployment workflows) 
- [x] Create workflows for deploying staging and production
  - We will deploy each APK build to Firebase Distribution
  - The build number will be increased automatically for each build

## Proof Of Work 📹

**Staging**

Example action: https://github.com/chornerman/Survey/actions/runs/3647291948

staging

**Production**

Example action: https://github.com/chornerman/Survey/actions/runs/3647292484

prod

### CODE REVIEW

**Comment by markgravity** [comment]
@chornerman what is the difference between #49 and #50? 🙏

**Comment by chornerman** [comment]
> @chornerman what is the difference between #49 and #50? 🙏

@markgravity #49 Is for setting up Firebase distribution, and #50 is for setting up Fastlane 😄 

If you have any suggestions, please let me know 🙏🏼

**Comment by markgravity** [comment]
> > @chornerman what is the difference between #49 and #50? 🙏
> 
> @markgravity #49 Is for setting up Firebase distribution, and #50 is for setting up Fastlane 😄
> 
> If you have any suggestions, please let me know 🙏🏼

Oh I see, but we don't plan setup for iOS also? 🙏

**Comment by chornerman** [comment]
> > > @chornerman what is the difference between #49 and #50? 🙏
> > 
> > 
> > @markgravity #49 Is for setting up Firebase distribution, and #50 is for setting up Fastlane 😄
> > If you have any suggestions, please let me know 🙏🏼
> 
> Oh I see, but we don't plan setup for iOS also? 🙏

@markgravity Yes, we plan to do it, but after completing the other core features of the app (confirmed with Trung). Please let me know if this sound okay to you 🙏🏼 

The reason behind this plan is to finish the core features of the app asap since I have 0 knowledge on IOS side 😅

**Review by markgravity — APPROVED** [review]
LGTM

**Comment by luongvo** [comment]
@chornerman rebasing issue here 👋

**Comment by luongvo** [comment]
@chornerman recall you here for rebasing 👋

**Comment by chornerman** [comment]
> @chornerman recall you here for rebasing 👋

Resolved 🙏🏼

File: android/.bundle/config [hunk 1]
```diff
@@ -0,0 +1,2 @@
+---
```
Comment (luongvo) [inline android/.bundle/config:1]: Same, Not sure why do we need this file?

**Review by chornerman — COMMENTED** [review]
@luongvo @markgravity I completely rewrite this PR using the same approach as @luongvo instead as it seems to be a lot better. Please help review it again and let me know if there are any issues. Sorry for the delay and thank you 🙏🏼

File: .github/workflows/decrypt_secret.sh [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+# ENV
+gpg --quiet --batch --yes --decrypt --passphrase="$LARGE_SECRET_PASSPHRASE" \
+  --output .env .github/workflows/secrets/.env.gpg
+
+gpg --quiet --batch --yes --decrypt --passphrase="$LARGE_SECRET_PASSPHRASE" \
+  --output .env.staging .github/workflows/secrets/.env.staging.gpg
+
+# Android
+gpg --quiet --batch --yes --decrypt --passphrase="$LARGE_SECRET_PASSPHRASE" \
+  --output android/app/google-services.json .github/workflows/secrets/google-services.json.gpg
```
[the remark this answers is not in the document]
Comment (luongvo) [reply .github/workflows/decrypt_secret.sh:16]: @chornerman If you combine all staging & production into 2 file and ignore pushing it in git so it's fine 👌

**Review by luongvo — APPROVED** [review]
lgtm

File: .github/workflows/decrypt_secret.sh (hunk 2 above)
[the remark this answers is not in the document]
Comment (luongvo) [reply .github/workflows/decrypt_secret.sh:16]: @chornerman I updated the above comment.

**Review by markgravity — APPROVED** [review]

**Pull request merged**